vpr/src/place/place.cpp

```diff
@@ -1350,6 +1359,32 @@ static e_move_result try_swap(float t,
 
     move_generator.process_outcome(move_outcome_stats);
 
+#ifdef VTR_ENABLE_DEBUG_LOGGING
+    
+    t_draw_state* draw_state = get_draw_state_vars();
+    if(f_placer_debug && draw_state->show_graphics){
+        std::string msg = available_move_types[0];
+        if(move_outcome == 0)
+            msg += vtr::string_fmt(", Rejected");
+        else if(move_outcome == 1)
+            msg += vtr::string_fmt(", Accepted");
+        else
+            msg += vtr::string_fmt(", Aborted");
+
+        msg += vtr::string_fmt(", Delta_cost: %1.6f (bb_delta_cost= %1.5f , timing_delta_c= %6.1e)", delta_c, bb_delta_c, timing_delta_c);
+        
+        auto& cluster_ctx = g_vpr_ctx.clustering();
+        
+        deselect_all();
+        draw_highlight_blocks_color(cluster_ctx.clb_nlist.block_type(blocks_affected.moved_blocks[0].block_num), blocks_affected.moved_blocks[0].block_num); 
+        draw_state->colored_blocks.clear();
+        
+        draw_state->colored_blocks.push_back(std::make_pair(blocks_affected.moved_blocks[0].old_loc, blk_GOLD));
+        draw_state->colored_blocks.push_back(std::make_pair(blocks_affected.moved_blocks[0].new_loc, blk_GREEN));
```

Comment:
Rather than reaching into  the drawing code and manipulating `draw_state` directly, I'd suggest wrapping these operations in an API. 

Generally we try to avoid coupling distince parts of the code base together to help maintainability. For instance, in this case, if someone ever refactors the drawing code and changes `draw_state`, they'll also need to update the placer (and vice versa). It also makes this the only code outside of `vpr/draw` which includes anything other than `draw.h`.

I'd suggest re-working this to put these operations behind and API in `draw.h` and removing the dependency on the other `draw_*.h` header files. For instance, by defining a function like:
```cpp
void set_draw_block_color(ClusterBlockId blk, ezgl::color color); //In draw.h
```

```cpp
void set_draw_block_color(ClusterBlockId blk, ezgl::color color) { //In draw.cpp
    t_draw_state* draw_state = get_draw_state_vars();
    draw_state->block_colors[blk] = color;
}
```

```cpp
//In place.cpp

ezgl::color OLD_BLOCK_LOCATION_COLOR = ezgl::GOLD;
ezgl::color NEW_BLOCK_LOCATION_COLOR = ezgl::GREEN;

set_block_color(blk, OLD_BLOCK_LOCATION);
set_block_color(blk, NEW_BLOCK_LOCATION);
//...
```

Reply:
Agreed, that makes sense.
I also think the code for DEBUG_LOGGING here is big enough that the whole straight-line code segment should be moved into one or a few subroutines and called.  That makes it easier for someone reading the code in the future to see high-level code like:
check_breakpoints (...);
draw_move (...);
That immediately gives a better idea of what's going on, and lets someone decide whether to read further into those routines or skip over them, depending on what he/she is trying to figure out.

Reply:
Mahshad I think you asked me what I meant by this.  I mean that the code guarded by DEBUG_LOGGING is large enough that it should be separated into a different subroutine.